pkg/asset/installconfig/azure/validation.go

```diff
@@ -188,11 +192,27 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 		if ultraSSDCapability == "" {
 			ultraSSDCapability = defaultUltraSSDCapability
 		}
+
+		capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+		}
+
 		if vmNetworkingType == "" {
-			vmNetworkingType = defaultVMNetworkingType
+			val, ok := capabilities[string(azure.AcceleratedNetworkingEnabled)]
```

Comment:
If there was an error, `capabilities` is going to be `nil` but you're still trying to access it here as a map. Down below you correctly skip instance validation if it's `nil`.

Reply:
Thanks, I'll add a check to guard against this.